client/finality-grandpa/src/authorities.rs

```diff
@@ -152,12 +167,43 @@ pub struct AuthoritySet<H, N> {
 	/// is lower than the last finalized block (as signaled in the forced
 	/// change) must be applied beforehand.
 	pending_forced_changes: Vec<PendingChange<H, N>>,
+	/// Track at which blocks the set id changes. This is useful when we need to prove finality.
```

Comment:
```suggestion
	/// Track at which blocks the set id changed. This is useful when we need to prove finality for a given block
	/// since we can figure out what set the block belongs to and when the set started/ended.
```